shared/src/androidMain/kotlin/com/devil/phoenixproject/util/DataBackupManager.android.kt

```
@@ -48,8 +59,43 @@ class AndroidDataBackupManager(private val context: Context, database: Vitruvian
+                    if (newFile != null) {
+                        context.contentResolver.openOutputStream(newFile.uri)?.use { outputStream ->
+                            tempFile.inputStream().use { it.copyTo(outputStream) }
+                        }
+                        tempFile.delete()
+                        Logger.d { "Session backup written to custom destination: ${destination.displayName}" }
+                        return
+                    }
+                }
+                tempFile.delete()
+                Logger.w { "Custom backup destination not writable, falling back to default" }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)